src/analytics/dsl.tsx

```diff
@@ -0,0 +1,12 @@
+export const dsl = {
+  "Follow Artist": {
+    intent: (object) => ({ artist_id: object.artist.id }),
+    success: (object) => ({ artist_id: object.artist.id }),
+    fail: (object) => ({ artist_id: object.artist.id }),
+  },
+  "Unfollow Artist": {
+    intent: (object) => ({ artist_id: object.artist.id }),
+    success: (object) => ({ artist_id: object.artist.id }),
+    fail: (object) => ({ artist_id: object.artist.id }),
+  },
+}
```

Comment:
Ohh I like this! (Also, such snazzy syntax)

I don't think I quite understand your idea of the multiple files though. There's one analytics file that actually sends the events and these are the methods that the decorator knows to access. E.g., send an event that this thing succeeded.

So, if we were to have multiple files for the different components etc, how would this context specific code end up in the analytics file?